File: apps/services/match/src/loaders/services.loader.ts
import { createCommandBus, createQueryBus } from "@zudojs/cqrs";
import type { CommandBus, QueryBus } from "@zudojs/cqrs";
import type { Container } from "@zudojs/container";
import { LOGGER_TOKEN } from "../constants/index.js";
import { registerMatchService } from "../services/index.js";

export interface ServiceBuses {
  readonly queryBus: QueryBus;
  readonly commandBus: CommandBus;
}

export function loadServices(container: Container): ServiceBuses {
  const queryBus = createQueryBus();
  const commandBus = createCommandBus();

  registerMatchService({ container, queryBus, commandBus });

  container.resolve(LOGGER_TOKEN).debug("Handlers registered", {
    queries: queryBus.size(),
    commands: commandBus.size(),
  });

  return { queryBus, commandBus };
}

File: services/analytics/src/betng_analytics/loaders/services_loader.py
from __future__ import annotations

from betng_service_kit import Container, QueryBus

from ..constants import LOGGER_TOKEN
from ..services import register_analytics_service


def load_services(container: Container) -> QueryBus:
    query_bus = QueryBus()

    register_analytics_service(container, query_bus)

    container.resolve(LOGGER_TOKEN).debug(
        "Query handlers registered", extra={"queries": query_bus.size()}
    )

    return query_bus
